app/models/spree/gateway/flow_io.rb

```diff
@@ -72,6 +76,12 @@ def create_profile(payment)
         create_flow_cc_profile!
       end
 
+      def credit(payment, credit_amount)
```

Comment:
Would it be a good idea to have an error of `"should be implemented in a sub-class"` in `Spree::Gateway`?

Reply:
Hmmm :thinking: we can discuss that option yes!! As far as I could see, the way it works is that the credit method is delegated to the to the provider [(reference code)](https://github.com/spree/spree/blob/2-3-stable/core/app/models/spree/gateway.rb#L3) . The problem we had was that the provider was configured for flow as itself, so we ended up in an infinite loop haha.

My main concern is that adding this method to the `Spree::Gateway` might affect other payments, so we will probably need to test this properly... Should I create a ticket for it?

Reply:
Agree with the tests! But if some other payment methods could be affected... that doesn't mean that issues should be happening right now?

Reply:
I think that it would depend on how the Gateway is configured. I will double check other Gateways we are using to be sure.

Reply:
The only other payment method we have today that is working with a similar configuration is Klarna. That gateway does not have a `credit` method set either; However, for some reason it is working ok... we tested with Ale and looks like when calling the `credit` method it actually calls the `refund` for some reason.

Adding the method to force the implementation sounds like a good idea to avoid these scenarios where the credit method is not implemented and still works like "magic". I will create a ticket to review this in the future :) thanks!